### ISSUE TITLE
Wayfire socket path generated is not unique across multiple users

### ISSUE THREAD

**Issue by marcusbritanicus** [WayfireWM/wayfire#2489]
**Describe the bug**
Currently, the wayfire socket path is created in the following manner:
```c++
const auto& dname  = wf::get_core().wayland_display;
std::string socket = pre_socket ?: "/tmp/wayfire-" + dname + ".socket";
```
When two instances of wayfire are started by two different users, both the users are given the same wayland-display name: `wayland-1`, resulting in the same socket name for both ipc instances.

**To Reproduce**
1. Start Wayfire on TTY1 (with IPC plugin enabled)
2. Switch to TTY 2, and start another instance of wayfire as **another user**
3. The value of wayland display is `wayland-1` for the instance in tty-2.
4. This causes IPC plugin to fail.

**Expected behavior**
The generated socket path should be unique and the IPC plugin should not fail to load in all instances of wayfire

**Suggested fix**
Instead of using `/tmp/`, it would be beneficial to use `XDG_RUNTIME_DIR` as in the case of wayland sockets. Alternatively, one could also append a random 4-digit string to the end of the name.

**Screenshots or stacktrace**
[wayfire.log](https://github.com/user-attachments/files/17270647/wayfire.log) Check lines 172-173.

**Wayfire version**
git

**Comment by ammen99** [comment]
Absolutely true, the /tmp file path was created mostly in the beginning to debug things and I have forgotten to change it. If someone sends a PR, it would be very welcome :)

**Comment by killown** [comment]
> Absolutely true, the /tmp file path was created mostly in the beginning to debug things and I have forgotten to change it. If someone sends a PR, it would be very welcome :)

you could easily add it for us :D

    pid_t pid = getpid();
    std::string socket = pre_socket ? pre_socket : "/tmp/wayfire-" + dname + "-" + std::to_string(pid) + ".socket";

**Comment by ammen99** [comment]
Fixed in #2490
